src/main/java/teammates/ui/webapi/action/UpdateInstructorPrivilegeAction.java

```diff
@@ -0,0 +1,141 @@
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.InstructorPrivilegeData;
+import teammates.ui.webapi.request.InstructorPrivilegeUpdateRequest;
+
+/**
+ * Update instructor privilege by instructors with instructor modify permission.
+ */
+public class UpdateInstructorPrivilegeAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.getId());
+
+        gateKeeper.verifyAccessible(
+                instructor, logic.getCourse(courseId), Const.ParamsNames.INSTRUCTOR_PERMISSION_MODIFY_INSTRUCTOR);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String courseId = getNonNullRequestParamValue(Const.ParamsNames.COURSE_ID);
+
+        String emailOfInstructorToUpdate = getNonNullRequestParamValue(Const.ParamsNames.INSTRUCTOR_EMAIL);
+        InstructorAttributes instructorToUpdate = logic.getInstructorForEmail(courseId, emailOfInstructorToUpdate);
+
+        if (instructorToUpdate == null) {
+            return new JsonResult("Instructor does not exist.", HttpStatus.SC_NOT_FOUND);
+        }
+
+        InstructorPrivilegeUpdateRequest request = getAndValidateRequestBody(InstructorPrivilegeUpdateRequest.class);
+
+        String sectionName = request.getSectionName();
+        String sessionName = request.getFeedbackSessionName();
+
+        Map<String, Boolean> courseLevelPrivilegesMap = request.getAllPresentCourseLevelPrivileges();
+        Map<String, Boolean> sectionLevelPrivilegesMap = request.getAllPresentSectionLevelPrivileges();
+        Map<String, Boolean> sessionLevelPrivilegesMap = request.getAllPresentSessionLevelPrivileges();
+
+        if (sectionName == null && sessionName == null) {
+            updateCourseLevelPrivileges(courseLevelPrivilegesMap, instructorToUpdate);
+            updateCourseLevelPrivileges(sectionLevelPrivilegesMap, instructorToUpdate);
+            updateCourseLevelPrivileges(sessionLevelPrivilegesMap, instructorToUpdate);
+        } else if (sessionName == null) {
+            updateSectionLevelPrivileges(sectionName, sectionLevelPrivilegesMap, instructorToUpdate);
+            updateSectionLevelPrivileges(sectionName, sessionLevelPrivilegesMap, instructorToUpdate);
+        } else {
+            updateSessionLevelPrivileges(sectionName, sessionName, sessionLevelPrivilegesMap, instructorToUpdate);
+        }
+
+        instructorToUpdate.privileges.validatePrivileges();
+        updateToEnsureValidityOfInstructorsForTheCourse(courseId, instructorToUpdate);
+
+        try {
+            logic.updateInstructor(
```

Comment:
Let's get the updated instructor.
```suggestion
            instructorToUpdate = logic.updateInstructor(
```